### PR TITLE
Es/bump jenkins version 2.222.4

### PR DESCRIPTION
**Pull request by mhenschke-atl**

### CODE REVIEW

File: acceptance-tests/pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
         <jenkins.acceptance-test-harness.version>1.89</jenkins.acceptance-test-harness.version>
         <bitbucket.version>7.8.0</bitbucket.version>
         <httpclient.version>4.5.10</httpclient.version>
-        <jenkins.version>2.204.6</jenkins.version>
+        <jenkins.version>2.222.4</jenkins.version>
```
Comment (mhenschke-atl) [inline acceptance-tests/pom.xml:20]: I did some poking around the plugin versions and they've all upgraded to 2.222.1- either Jenkins isn't using semver or there's some other reason they're not going higher. I don't think it makes a huge difference, but we might want to go 2.222.1 just for consistency

Comment (dkjellin) [reply acceptance-tests/pom.xml:20]: I think most plugins, like us, try the lowest possible version. I think Jenkins is _mostly_ semver but not always so it is safer to play it low on versions. I say if this works, let's stay on `2.222.4`

File: acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/client/JenkinsApplinksClient.java [hunk 2]
```diff
@@ -16,18 +20,20 @@
...
         OAuthConsumer consumer = newConsumer();
-        String consumerCreateUri = absoluteUrl(CREATE_OAUTH_CONSUMER_PATH);
-        RestAssured.given()
+        String consumerCreateUri = jenkinsUtils.toAbsoluteUrl(CREATE_OAUTH_CONSUMER_PATH);
+
+        RequestSpecification request = RestAssured.given()
                 .contentType(URLENC)
-                .formParam("json", jsonSerializer.writeValueAsString(consumer))
+                .formParam("json", jsonSerializer.writeValueAsString(consumer));
+        jenkinsUtils.decorateWithCookie(request)
```
Comment (dkjellin) [inline acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/client/JenkinsApplinksClient.java:36]: I'm thinking maybe you could move things around a bit to make it flow a bit differently (and better?)
something like
```java

RequestSpecification request =  jenkinsUtils.decorateWithCookie(RestAssured.given())
                    .contentType(URLENC)
                    .formParam("json", jsonSerializer.writeValueAsString(consumer))
                    .formParam("json", jsonSerializer.writeValueAsString(consumer))
                    .when()
                    .post(consumerCreateUri)
                    .then()     
                    .statusCode(302);
```

From my understanding that should still work just as well.

File: acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/client/JenkinsApplinksClient.java [hunk 3]
```diff
@@ -16,18 +20,20 @@
     private static final String CREATE_OAUTH_CONSUMER_PATH = "/bbs-oauth/create/performCreate";
 
     private final ObjectMapper jsonSerializer = new ObjectMapper();
-    private final String baseUrl;
+    private final JenkinsUtils jenkinsUtils;
 
     public JenkinsApplinksClient(String baseUrl) {
-        this.baseUrl = removeEnd(baseUrl, "/");
+        jenkinsUtils = new JenkinsUtils(removeEnd(baseUrl, "/"));
```
Comment (dkjellin) [inline acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/client/JenkinsApplinksClient.java:26]: you change from `removeStart` to `removeEnd` I assume there is a reason for this? `ermoveEnd` seems more correct to me, but I'm curious.

**Review by dkjellin — APPROVED** [review]

File: acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/util/JenkinsUtils.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    private final String baseUrl;
+
+    private static final String CRUMB_ISSUER_URL = "/crumbIssuer/api/xml";
+
+    public JenkinsUtils(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    public RequestSpecification decorateWithCookie(RequestSpecification request) {
```
Comment (EliSalter-atl) [inline acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/util/JenkinsUtils.java:19]: I think we could we rename this method to also indicate that we are decorating with a Jenkins crumb too

**Review by EliSalter-atl — APPROVED** [review]

**Pull request merged**